crates/configuration/src/version3/mod.rs

```diff
@@ -714,14 +714,48 @@ fn convert_scalar_types(
                             .cloned()
                             .unwrap_or(BTreeMap::new()),
 
-                        type_representation: representations.0.get(&t).cloned(),
+                        type_representation: convert_or_infer_type_representation(
+                            representations.0.get(&t).cloned(),
+                            &t,
+                        ),
                     },
                 )
             })
             .collect(),
     )
 }
 
+/// Infer scalar type representation from scalar type name, if necessary. Defaults to JSON representation
+fn convert_or_infer_type_representation(
+    representation: Option<query_engine_metadata::metadata::TypeRepresentation>,
+    scalar_type_name: &ScalarTypeName,
+) -> query_engine_metadata::metadata::TypeRepresentation {
+    if let Some(representation) = representation {
+        representation
+    } else {
+        match scalar_type_name.as_str() {
```

Comment:
We already have the `default_base_type_representations` to do this mapping from scalar type name to `TypeRepresentation`.  If this function adds any more mappings we should add them there rather than adding a new layer of indirection. The idea that whatever mappings are in the configuration are what will be returned in the schema is a useful property for understanding all this, and we would need a compelling reason to remove that property.

Reply:
v0.2 requires type representation in schema responses, however type representation is optional in the configuration itself.

So this is a piece of backwards compatiblity code, to allow for configurations that are missing type representation.

We could instead create configuration v6, and refuse to run with older configuration versions.

As for the use of `default_base_type_representations`: this mapping only exists for configurations v4 and v5. Here in v3 we replicate that functionality instead.
I prefered copying the code but keeping the different modules separate.

And yes it's a bit cheaty: we have a default when creating a new config, and reuse that default when type representation is missing.

Reply:
It's probably best to leave version 3 configs doing whatever they were before, ie falling back to JSON where they have no `TypeRepresentation`, or where they used the deprecated `Number` or `Integer` types.

Also, there is no need for a version 6, it looks like we were already making any mention of `Number` and `Integer` fallback to JSON before this PR, and since we have our own type exposed in the config type, we won't be removing any items and thus making any breaking changes.

Reply:
Defaulting to JSON is a behavior change. The previous behavior was that since representation was optional in both schema response and configuration, we just returned whatever was in configuration.

This applies to applies to configurations 3,4,5

Confirm we'd rather default to JSON if representation missing, without first trying to infer based on type name?

Reply:
Yes, default to JSON, this is what is assumed if no type representation is provided, so semantically it's no change, it just means the connector chooses JSON instead of engine doing it.